src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_LENGTH = 10;
+    private int totalCount;
+    private int filledCount;
+    private Object[] items;
+
+    public ArrayList() {
+        items = new Object[INITIAL_LENGTH];
+        totalCount = items.length;
+        filledCount = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (isFull()) {
+            grow();
+        }
+        items[filledCount] = value;
+        filledCount += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != filledCount) {
+            checkOutOfBoundsException(index);
+        }
 
+        if (isFull()) {
+            grow();
+        }
```

Comment:
put that check inside of grow - rename it to growIfNeeded to better reflect its logic